src/components/composer/composer/composer.ts

```diff
@@ -495,12 +536,55 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
           break;
       }
       if (this.env.model.getters.showSelectionIndicator() && end === start && end === token.end) {
-        result[result.length - 1].class = SelectionIndicatorClass;
+        result[result.length - 1].class = selectionIndicatorClass;
       }
     }
     return result;
   }
 
+  /**
+   * Split an array of HTMLContents into lines. Each NEWLINE character encountered will create a new
+   * line. Contents can be split into multiple parts if they contain multiple NEWLINE characters.
+   */
+  private splitHtmlContentIntoLines(contents: HtmlContent[]): HtmlContent[][] {
```

Comment:
We need to split the tokens into lines _somewhere_, to create a new paragraph for each line.

I do it here rather than in the ContentEditableHelper so it's possible to test it.